src/TriMesh.cpp

```diff
@@ -168,8 +168,8 @@ bool TriMesh::collapse_edge(const Tuple& loc0, std::vector<Tuple>& new_edges)
     size_t test_fid2 = loc1.get_fid();
 
     assert(
-        ((std::count(n12_intersect_fids.begin(), n12_intersect_fids.end(), n1_fids) &&
-          std::count(n12_intersect_fids.begin(), n12_intersect_fids.end(), n2_fids)),
+        ((std::count(n12_intersect_fids.begin(), n12_intersect_fids.end(), n1_fids.size()) &&
+          std::count(n12_intersect_fids.begin(), n12_intersect_fids.end(), n2_fids.size())),
```

Comment:
is this correct?

Reply:
This is quite unclear, `std::count` does not require boolean

Reply:
I just fixed compilation, @yfzhou42 is this correct?

Reply:
it should be the fids of the two faces along the same edge. Fixed

Reply:
asserting it is not zero. I want to say n12_intersect_fids contains test_fid1 and test_fid2